### ISSUE TITLE
Redesign Rancher buttons

### ISSUE THREAD

**Issue by edenhernandez-suse** [rancher/dashboard#8048]
Rancher uses a set of buttons with several issues. The hierarchy is not clear, there are no rules around the Secondary and Tertiary, and the colour scheme is both outdated (uses the old Rancher branding colour) and lacks some contrast (especially the Tertiary button). 

We need a new set of buttons with clear differences between each type of button, defined purposes to avoid ambiguity or confusion and the right contrast.

This work will be part of the [Accessibility Epic: better contrast in elements in both light and dark modes](https://github.com/rancher/dashboard/issues/4386)

**Comment by edenhernandez-suse** [comment]
[Buttons options](https://xd.adobe.com/view/bf02de5f-7d2c-4a79-b297-75a464f31556-ea51/)

**Comment by edenhernandez-suse** [comment]
Final proposal: [New buttons Xd link](https://xd.adobe.com/view/a55aeee2-6b39-4780-832b-37eacc409ec4-f009/)

@kwwii @manuelecarlini @bisht-richa @scures any feedback is welcome

**Comment by edenhernandez-suse** [comment]
[Proposal](https://xd.adobe.com/view/a55aeee2-6b39-4780-832b-37eacc409ec4-f009/)

@kwwii had comments about the Tertiary button:

![image](https://user-images.githubusercontent.com/120640736/219346765-a1882dbd-904e-4999-8f29-e84d91136333.png)

> I think I was trying to understand how a user will identify the Tertiary button as a button without moving their mouse over them if it's only text with normal text colour in the default state.


The point here, IMO, is that colour alone is not enough to make something "clickable" (some users won't see the colour). Generally speaking, we need to be sure that an action is clear using context and proper labelling. My assumption was that we can make Tertiary buttons explicit enough just with those two concepts (context and labelling), even if it's a text-only button.

However, we need to test this and see if it doesn't work in all scenarios. So far, the only one I've found where a Tertiary button could be useful is this table action:

![image](https://user-images.githubusercontent.com/120640736/219348096-a423f255-76df-4a84-b6ff-ebe422c6c6ef.png)

![image](https://user-images.githubusercontent.com/120640736/219348138-6121781d-b4d6-48a1-86ea-47bdf0492ff8.png)

If text-only buttons don't work for us, we can make an alternative (like the ghost button that is currently used).
